### ISSUE TITLE
How to obtain the initial weight in Keras?

### ISSUE THREAD

**Issue by sekti92** [keras-team/keras#11963]
Good Evening,

I was following tutorial in here: https://sefiks.com/2018/08/06/deep-face-recognition-with-keras/
I would like to know how to show the initial weight of pretrained model before and after the training?

As far as I know In tensorflow I can show **weight from input layer to hidden layer** and **weight from hidden layer to output layer**

This is how I obtain weight in Tensorflow

> data = np.array(attributes, 'int64')
> target = np.array(labels, 'int64')
> 
> feature_columns = [tf.contrib.layers.real_valued_column("", dimension=2, dtype=tf.float32)]
> learningRate = 0.1
> epoch = 10000
> 
> // https://www.tensorflow.org/api_docs/python/tf/metrics
> validation_metrics = {
>     "accuracy": tf.contrib.learn.MetricSpec(metric_fn = tf.contrib.metrics.streaming_accuracy ,
>     prediction_key = tf.contrib.learn.PredictionKey.CLASSES),
>     "precision": tf.contrib.learn.MetricSpec(metric_fn = tf.contrib.metrics.streaming_precision ,
>     prediction_key = tf.contrib.learn.PredictionKey.CLASSES),
>     "recall": tf.contrib.learn.MetricSpec(metric_fn = tf.contrib.metrics.streaming_recall ,
>     prediction_key = tf.contrib.learn.PredictionKey.CLASSES),
>     "mean_absolute_error": tf.contrib.learn.MetricSpec(metric_fn = tf.contrib.metrics.streaming_mean_absolute_error ,
>     prediction_key = tf.contrib.learn.PredictionKey.CLASSES),
>     "false_negatives": tf.contrib.learn.MetricSpec(metric_fn = tf.contrib.metrics.streaming_false_negatives ,
>     prediction_key = tf.contrib.learn.PredictionKey.CLASSES),
>     "false_positives": tf.contrib.learn.MetricSpec(metric_fn = tf.contrib.metrics.streaming_false_positives ,
>     prediction_key = tf.contrib.learn.PredictionKey.CLASSES),
>     "true_positives": tf.contrib.learn.MetricSpec(metric_fn = tf.contrib.metrics.streaming_true_positives ,
>     prediction_key = tf.contrib.learn.PredictionKey.CLASSES)
> }

> // validation monitor
> validation_monitor = tf.contrib.learn.monitors.ValidationMonitor(data, target, every_n_steps=500, 
>     metrics = validation_metrics)
> 
> classifier = tf.contrib.learn.DNNClassifier(
>     feature_columns = feature_columns,
>     hidden_units = [3],
>     activation_fn = tf.nn.sigmoid,
>     optimizer = tf.train.GradientDescentOptimizer(learningRate), 
>     model_dir = "model",
>     config = tf.contrib.learn.RunConfig(save_checkpoints_secs = 1)
> )
> 
> classifier.fit(data, target, steps = epoch,
>     monitors = [validation_monitor])
> 
> // print('Params:', classifier.get_variable_names())
> '''
> Params: ['dnn/binary_logistic_head/dnn/learning_rate', 'dnn/hiddenlayer_0/biases', 'dnn/hiddenlayer_0/weights', 'dnn/logits/biases', 'dnn/logits/weights', 'global_step']
> '''
> 
> print('total steps:', classifier.get_variable_value("global_step"))
> print('weight from input layer to hidden layer: ', classifier.get_variable_value("dnn/hiddenlayer_0/weights"))
> print('weight from hiddent layer to output layer: ', classifier.get_variable_value("dnn/logits/weights"))
> 

How to do that in Keras? Thank you.

**Comment by msymp** [comment]
Hello @sekti92 , here is a Stack Overflow post that addresses your query about how to obtain initial weights in Keras:
https://stackoverflow.com/questions/52703096/calculate-visualize-tensorflow-keras-dense-model-layer-relative-connection-weigh
Please specifically study _Srilekha Vinjamara's_ answer in the post, and write your code and run it on Keras with TensorFlow backend. Please let us know. Thanks.

**Comment by sekti92** [comment]
@msymp hi thanks, for remind me
